src/main/java/net/sourceforge/pmd/util/fxdesigner/DesignerStarter.java

```diff
@@ -63,14 +65,21 @@ private static MainCliArgs readParameters(String[] argv) {
 
     }
 
-
+    /**
+     * Starting from PMD 7.0.0 this method usage will be limited for development.
+     * CLI support will be provided by pmd-cli
+     */
+    @Deprecated
+    @InternalApi
     public static void main(String[] args) {
```

Comment:
I'm not sure we should deprecate this. If there is no main here then it's harder to launch because people also need to build `pmd-cli`. Maybe it should just be documented, that this is not the main used by `pmd designer` after pmd/pmd#4059?

Reply:
yeah, the goal was simply to mark it as internal… this main should ONLY be used during designer development

The deprecation is intended to mark it (if anyone was using it somewhere?), not necessarily to mark it for removal.

Reply:
Ok, that makes sense. We should probably make that explicit in the `@deprecated` javadoc section

Reply:
Alternatively, I may introduce a new class `net.sourceforge.pmd.util.fxdesigner.internal.Main` with just a main invoking `DesignStarter`, but being part of an internal package; and actually get this one flagged for removal in the future. WDYT?

Reply:
I think it's easier to have the main here and marked `@InternalApi`. The annotation was made for this kind of case after all, and it is orthogonal to the `@Deprecated` annotation. I think maybe the simplest solution is not to deprecate but to simply use `@InternalApi`? Using a separate class for the main seems to me like it would complexify the entry point for an unclear benefit.